### DISCUSSION TITLE
handling flutter views in native apps

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#16531] [Q&A] [answered]
My problem is that i have a native app (iOS and Android) which also have some views made by flutter.
I think i have to use 2 drivers in one session. One native driver (iOS/Android) and a flutter driver so that i can dynamically switch the driver when i want to interact with the native elements or the flutter elements.

I can use 2 drivers in one session when i use 2 simulators/emulators but when i try to have 2 drivers on one simulator, the first session is gone after the 2nd driver is initialized. I thought its enough to set different localPorts and wdaLocalPorts but no luck until now.

So i’m not 100% sure if it is possible with appium to use 2 drivers on one emulator/simulator.

**Comment by jlipps** [comment, marked as answer]
That shouldn't be necessary, since the flutter driver includes the uiautomator2 driver and xcuitest driver. I've never used the flutter driver myself, maybe @KazuCocoa can advise.

**Reply by KazuCocoa** [reply]
https://github.com/appium-userland/appium-flutter-driver includes XCUITest/UIA2 driver by switching contexts. The driver may help.

**Reply by RSM-SBA** [reply]
ahhh thats great. Thank you for your response.
